dealii-policy-compatibility.md

```diff
@@ -0,0 +1,43 @@
+|**M7.** Come with an open source (BSD style) license. |Full| LGPL. |
+|**M8.** Provide a runtime API to return the current version number of the software. |Full| Via `Utilities::dealii_version_string()`. |
+|**M9.** Use a limited and well-defined symbol, macro, library, and include file name space. | Mostly| All header files are prefixed with `deal.II/`. All C++ names are in a namespace `dealii::`. (The exception of a few overloads of `std::pow`, `std::sin`, etc with arguments that are classes in namespace `dealii` -- consequently, these overloads cannot clash with other libraries.) Almost all macros are prefixed with `DEAL_II_`. The exception are a few macros for exception-related things; namely, `Assert`, `AssertThrow`, `AssertNothrow`, and a set of macros with names `DeclException*`. |
+|**M10.** Provide an xSDK team accessible repository (not necessarily publicly available). |Full| https://github.com/dealii/dealii |
+|**M11.** Have no hardwired print or IO statements that cannot be turned off. |Full| Yes. All logging is off by default. |
+|**M12.** For external dependencies, allow installing, building, and linking against an outside copy of external software. |Full| Yes. This is true for all dependencies. We bundle BOOST, TBB, and UMFPACK, but cmake will use an external version if one can be found. |
+|**M13.** Install headers and libraries under \<prefix\>/include and \<prefix\>/lib. |Full| Yes. The install path is chosen via `CMAKE_INSTALL_PREFIX`. |
+|**M14.** Be buildable using 64 bit pointers. 32 bit is optional. |Full| We make no assumptions. Whatever the compiler wants to use. |
+|**M15.** All xSDK compatibility changes should be sustainable. |Full| Yes. Rather than providing the xSDK options as add ons, we have incorporated them fully into the basic deal.II infrastructure, so that xSDK options get the same maintaince and support as any other feature. |
+|**M16.** The package must support production-quality installation compatible with the xSDK install tool and xSDK metapackage. |Full| Yes. deal.II is supported by SPACK. |
+
+
+### Recommended Policies
+
+| Policy                 |Support| Notes                   |
+|------------------------|-------|-------------------------|
+|**R1.** Have a public repository. |Full | http://github.com/dealii/dealii |
+|**R2.** Possible to run test suite under valgrind in order to test
+|for memory corruption issues. | Work in progress | Not at the moment, but we are working on it. |
+|**R3.** Adopt and document consistent system for error conditions/exceptions. | Yes | Yes, we have a system of macros that is uniformly used for this. Errors either lead to termination of the program if, for example, a function is called with invalid arguments (but this can be switched to reporting C++ exceptions). All other error conditions lead to throwing C++ exceptions. |
+|**R4.** Free all system resources acquired as soon as they are no longer needed. |Full| Yes, that's the goal. |
```

Comment:
Does this account for `VectorMemory` and friends? I would argue that we do not release system resources since we keep vectors around, though this point is pedantic.

Reply:
Hm, good question. What should we do about it? We keep these vectors around because we typically do need them again.

Reply:
I think it is worth mentioning that `GrowingVectorMemory<VectorType>` caches temporary vectors between calls to a Krylov solver.

Reply:
I've updated that as well.